### ISSUE TITLE
Errors on site without expected parameters set

### ISSUE THREAD

**Issue by paskal** [jnjosh/internet-weblog#9]
Hello!

$subj, when trying to run this theme against my [blog](https://github.com/paskal/terrty/tree/master/blog) source I get following:

```
Building sites … ERROR 2018/03/20 20:35:50 Error while rendering "section" in "": template: theme/_default/list.html:2:3: executing "theme/_default/list.html" at <partial "page_header...>: error calling partial: template: theme/partials/page_header.html:5:10: executing "theme/partials/page_header.html" at <$.Scratch.Add>: error calling Add: Can't apply the operator to the values
ERROR 2018/03/20 20:35:50 Error while rendering "home" in "": template: theme/index.html:2:3: executing "theme/index.html" at <partial "page_header...>: error calling partial: template: theme/partials/page_header.html:5:10: executing "theme/partials/page_header.html" at <$.Scratch.Add>: error calling Add: Can't apply the operator to the values
ERROR 2018/03/20 20:35:50 Error while rendering "taxonomy" in "": template: theme/_default/list.html:2:3: executing "theme/_default/list.html" at <partial "page_header...>: error calling partial: template: theme/partials/page_header.html:5:10: executing "theme/partials/page_header.html" at <$.Scratch.Add>: error calling Add: Can't apply the operator to the values
ERROR 2018/03/20 20:35:50 Error while rendering "taxonomyTerm" in "": template: theme/_default/list.html:2:3: executing "theme/_default/list.html" at <partial "page_header...>: error calling partial: template: theme/partials/page_header.html:5:10: executing "theme/partials/page_header.html" at <$.Scratch.Add>: error calling Add: Can't apply the operator to the values

```

The reason is likely some absent `.Site.` parameter but you can't tell it from error message, I guess input parameters handling could be improved. https://github.com/cboettig/hugo-now-ui/issues/9 might be used as reference point.

**Comment by paskal** [comment]
https://github.com/hivickylai/hugo-theme-sam/pull/3 is an example of how this can be fixed.

**Comment by digitalcraftsman** [comment]
Hello @paskal,

due to this issue the generation of a demo for your theme at https://themes.gohugo.io/ failed. Recently, we began a "spring cleaning" and are about to remove themes that fail to build and / or are unmaintained.

It would be great to see your theme added again. To do so please let us know when the issue has been resolved. For further information please have a look at https://github.com/gohugoio/hugoThemes/pull/479.

**Comment by paskal** [comment]
@digitalcraftsman, it's not my theme. But if you came there anyway, please consider merging my changes to your themes, which I've prepared some time ago and which got zero attention so far:
https://github.com/digitalcraftsman/hugo-cactus-theme/pull/52
https://github.com/digitalcraftsman/hugo-creative-theme/pull/40
https://github.com/digitalcraftsman/hugo-icarus-theme/pull/119
https://github.com/digitalcraftsman/hugo-minimalist-theme/pull/14
https://github.com/digitalcraftsman/hugo-steam-theme/pull/25
https://github.com/digitalcraftsman/hugo-strata-theme/pull/49
https://github.com/digitalcraftsman/hugo-type-theme/pull/33

**Comment by jnjosh** [comment]
Fixed in 85ddc72
